.github/workflows/nim_test.yml

```diff
@@ -0,0 +1,33 @@
+---
+name: nim_test
+
+# yamllint disable-line rule:truthy
+on:
+  workflow_dispatch:
+  push:
+    branches:
+      - master
+  pull_request:
+
+jobs:
+  nim_test:
+    runs-on: ${{matrix.os}}
+    strategy:
+      fail-fast: false
+      matrix:
+        os:
+          - ubuntu-latest
+        nim-version:
+          - '1.6.12'
```

Comment:
`stable` changes over time. My suggestion is to add new versions when needed (that is why I have introduced the `strategy/matrix`).

Reply:
> stable changes over time.

Exactly. Isn't it the whole point of building everything automatically?

Following stable here provides early warning and ensures there's no big migration event ever.

Reply:
Imagine you change some file and make a PR. And at the same time `stable` changes. It might turn out that on your PR suddenly many other, untouched files will fail.

I think it might be easier to simply add a new version from time to time.

Reply:
I think it might be a good idea adding another workflow doing a build every week or month with `stable` version. Something like:
```yaml
---
name: nim_stable_test

# yamllint disable-line rule:truthy
on:
  workflow_dispatch:
  cron: '3 1 * * 1'

jobs:
  nim_test:
    runs-on: ubuntu-latest

    steps:
      - uses: actions/checkout@v3

      - uses: jiro4989/setup-nim-action@v1

      - name: Display nim version:
        run: |
          nim --version

      - name: Build and test
        run: |
          ./check_all_nim_files.sh
...
```

Reply:
> Imagine you change some file and make a PR. And at the same time stable changes. It might turn out that on your PR suddenly many other, untouched files will fail.

If other packages fail, it's not a problem, people can notice and start upgrading them right away.

Automation should help manual reviewing, not replace it. Checking PRs on stable helps ensuring the contents of the PR actually build, and do it while the project is in the attention scope of the contributor, so they can make the necessary fixes.

This repo is not some critical piece of software which needs to be automatically deployed from the pipeline, there's no need to avoid possible minor breakage here or there at all costs. Moreover, stable is not as volatile, breaking changes are usually added for a reason.

Reply:
I am in favor of using stable now and fall back to manual versioning if we encounter problems.
In nim-lang/bigints, we removed automatic versioning in https://github.com/nim-lang/bigints/pull/97 to explicitly state the versions supported.

Here, we do not need backwards compatibility.

Reply:
The argument of tests failing on units not related to the PR at hand is solid. It all boils down to what do we want from this job.

1. If we want it to be a hard-stop on any fail ("all green checks -> go!"), than it has to not fail due to underlying compiler/stdlib updates (even though the probability of it happening is not that great) and hard-coding some version is necessary.
2. If we want it to be just a thing to consider, an assistance for manual reviews, than `stable` is definitely better.

Having thought about it a bit more, I don't think I have a strong opinion on the matter now.

Reply:
@ZoomRmc I would suggest 1 (think about the hacktoberfest madness) and the cyclically running workflow with `stable` and `devel`.